### PR TITLE
Changes for rename from gnatsd -> nats-server

### PR DESCRIPTION
**Pull request by derekcollison**
These are various changes needed for the repo and binary rename.

FOSSA will be an issue for @caniszczyk 

Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,8 +3,12 @@ module github.com/nats-io/gnatsd
 require (
 	github.com/nats-io/go-nats v1.7.2
 	github.com/nats-io/jwt v0.2.4
+	github.com/nats-io/nats-server v1.4.1
```
Comment (kozlovic) [inline go.mod]: Something weird here... why go.mod of nats-server would require nats-server v.1.4.1? Also top of this file still says /nats-io/gnatsd.

Comment (derekcollison) [reply go.mod]: Will take a look. I removed it from vendor and made sure that worked..

**Comment by derekcollison** [comment]
Will fix crosscompile. Eventually go to mod only but for now while we build with 1.11.x use vendor IMO.

File: docker/Dockerfile.alpine [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+WORKDIR $GOPATH/src/github.com/nats-io/nats-server
 
 MAINTAINER Waldemar Quevedo <wally@synadia.com>
 
 RUN apk add --update git
 
 COPY . .
 
-RUN CGO_ENABLED=0 go build -v -a -tags netgo -installsuffix netgo -ldflags "-s -w -X github.com/nats-io/gnatsd/server.gitCommit=`git rev-parse --short HEAD`" -o /nats-server
+RUN CGO_ENABLED=0 go build -v -a -tags netgo -installsuffix netgo -ldflags "-s -w -X github.com/nats-io/nats-server/server.gitCommit=`git rev-parse --short HEAD`" -o /nats-server
```
Comment (kozlovic) [inline docker/Dockerfile.alpine]: `GO111MODULE=off` here?

Comment (derekcollison) [reply docker/Dockerfile.alpine]: yes fixed.

**Pull request merged**